### PR TITLE
[driver] TMP12x temperature sensor

### PR DESCRIPTION
**Pull request by chris-durand**
Simple driver for TMP121/TMP123/TMP125 temperature sensors. The driver has been in use for some time on a custom board with an STM32F042K6. I'll test the Nucleo example tomorrow.

### CODE REVIEW

**Comment by rleh** [comment]
The only difference between TMP121 und TMP123 are the swapped GND and NC pins, or am I missing something?

File: src/modm/driver/temperature/tmp12x.hpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	}
+
+	modm::ResumableResult<Temperature>
+	read()
+	{
+		RF_BEGIN();
+
+		RF_WAIT_UNTIL(this->acquireMaster());
+		Cs::reset();
+		SpiMaster::setDataMode(SpiMaster::DataMode::Mode0);
```
Comment (rleh) [inline src/modm/driver/temperature/tmp12x.hpp]: This should be done in the [`ConfigurationHandler`](https://docs.modm.io/develop/api/stm32f042t6y6/classmodm_1_1SpiDevice.html), like e.g. in the the [DRV832x driver](https://github.com/modm-io/modm/blob/02660d04bf633ae619daa52ce5e19d24e546ede9/src/modm/driver/motor/drv832x_spi_impl.hpp#L22-L25).

File: src/modm/driver/temperature/tmp12x.hpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+			Cs::set();
+		}
+
+		RF_END_RETURN(Temperature(buffer_[1] | (buffer_[0] << 8)));
+	}
+private:
+	std::array<uint8_t, 2> buffer_{};
+};
+
+/**tmp12x_test
```
Comment (rleh) [inline src/modm/driver/temperature/tmp12x.hpp]:
```suggestion
/**
```

**Review by rleh — COMMENTED** [review]
Nice driver 👍🏽

**Comment by chris-durand** [comment]
> The only difference between TMP121 und TMP123 are the swapped GND and NC pins, or am I missing something?

Most likely the only difference is the chip marking. For the ones I had the NC pins were bonded to GND.

File: src/modm/driver/temperature/tmp12x.hpp (hunk 1 above)
Comment (chris-durand) [reply src/modm/driver/temperature/tmp12x.hpp]: Shouldn't we also set the data size? If a DRV832x was shared on a bus with another device with a different setting the configuration wouldn't be reset to 8 bit.

Comment (rleh) [reply src/modm/driver/temperature/tmp12x.hpp]: Yes.

Comment (chris-durand) [reply src/modm/driver/temperature/tmp12x.hpp]: A lot of SPI drivers which use the default settings don't have configuration handlers at all. We should fix that at some point.

**Review by rleh — APPROVED** [review]

File: src/modm/driver/temperature/tmp12x.hpp (hunk 1 above)
Comment (rleh) [reply src/modm/driver/temperature/tmp12x.hpp]: Absolutely! Somebody™️ just has to do (and test) it...

**Review by salkinium — APPROVED** [review]
Very nice!

**Comment by chris-durand** [comment]
Please don't merge until I have tested the example.

**Comment by chris-durand** [comment]
I have tested the example and it works.

**Pull request merged**